### ISSUE TITLE
[Question] Postprocessor return type

### ISSUE THREAD

**Issue by OS-JaR** [fonimus/ssh-shell-spring-boot#156]
Hi all,

first of all, I'd like to thank you for this procject, it seems to be a very useful continuation of springs shell project.
But after using it a few days, I start wondering why the [PostProcessor interface](https://github.com/fonimus/ssh-shell-spring-boot/blob/master/starter/src/main/java/com/github/fonimus/ssh/shell/postprocess/PostProcessor.java) does not allow other return types as `String`. Is there any reason, espacially since the [TypePostProcessorResultHandler](https://github.com/fonimus/ssh-shell-spring-boot/blob/master/starter/src/main/java/com/github/fonimus/ssh/shell/postprocess/TypePostProcessorResultHandler.java) handle results as `Object`. DOn't you think, that it would be much easier to allow Objects as return Type or maybe even better something generic like 

```
public interface PostProcessor<TIn, TOut> {

    String getName();

    TOut process(TIn result, List<String> parameters) throws PostProcessorException;
}
```

With this change it would be a lot easier to chain post processors, since it's currently not possible to do stuff like:
`my-command parameter1 parameter2 | postprocessor1 returnAsDate | postprocessor2readDateAndConvert2OtherTimezone` 

BR,
JaR

**Comment by OS-JaR** [comment]
Hi François,

thanks alot for yout effort, it seems this works like charm, so this feature can be merged into master/1.7.0 :)
Are there any future plans for additional features (just curiosity)?

Thanks
JaR

**Comment by fonimus** [comment]
I released 1.7.0 with this new feature ;)

For now no plans for new features that I can think of, but new ideas are always welcome !
